openhands/sdk/conversation/state.py

```diff
@@ -118,6 +121,17 @@ def secrets_manager(self) -> SecretsManager:
         """Public accessor for the SecretsManager (stored as a private attr)."""
         return self._secrets_manager
 
+    def set_on_state_change(
+        self, callback: Callable[[str, Any, Any], None] | None
+    ) -> None:
+        """Set a callback to be called when state changes.
+
+        Args:
+            callback: A function that takes (field_name, old_value, new_value)
+                     or None to remove the callback
+        """
+        self._on_state_change = callback
+
```

Comment:
@OpenHands
We should make sure this on_state_change has the same type as `ConversationCallbackType` and we should just create ConversationStateUpdateEvent inside this function.

Instead of creating separate callbacks, we should be able to just put `conversation._on_event` as callback here